dv/cosim/cosim.h

```diff
@@ -110,6 +110,12 @@ class Cosim {
   // A full 64-bit value is provided setting both the mcycle and mcycleh CSRs.
   virtual void set_mcycle(uint64_t mcycle) = 0;
 
+  // Set the value of a CSR. This is only used for MHPMCOUNTER3..12 registers
```

Comment:
I'd ditch the stuff explaining it's only used for performance counters and why that's needed as this is meant to be a generic interface. This function can set any CSR if it wants and maybe some future use of this interface will have a need to do it for other CSRs.

Perhaps add a more generic explanation, e.g. 'This is useful to keep CSRs in sync between the DUT and the model where their values cannot be easily modeled, e.g. performance counters'